### PR TITLE
fix typed linting

### PR DESCRIPTION
**Pull request by patricklx**
fixes https://github.com/ember-cli/eslint-plugin-ember/issues/2062

* use ts own parser
* only patch when its actually called
* note that ts files also need to use gjs-gts-parser to handle gts imports

### CODE REVIEW

File: test-projects/gts/src/menu.gts [hunk 1]
```diff
@@ -62,7 +61,7 @@ const PlainTrigger: TOC<{
   <@menu.Button {{@trigger}} ...attributes>
     {{yield @menu}}
   </@menu.Button>
-</template>;
+</template> as unknown;
```
Comment (NullVoxPopuli) [inline test-projects/gts/src/menu.gts:64]: Why so much unknown casting?

File: test-projects/gts/package.json [hunk 2]
```diff
@@ -2,14 +2,22 @@
...
+    "@glint/template": "^1.3.0",
+    "@glimmer/tracking": "^1.1.2",
+    "@glimmer/component": "^1.1.2",
     "@typescript-eslint/eslint-plugin": "^6.21.0",
     "@typescript-eslint/parser": "^6.21.0",
     "ember-eslint-parser": "workspace:^",
+    "ember-source": "^5.6.0",
     "eslint": "^8.0.1",
     "eslint-plugin-ember": "^12.0.0",
+    "limber": "^0.0.1",
```
Comment (NullVoxPopuli) [inline test-projects/gts/package.json:19]: This isn't related, we should remove this dependency. Last publish was 10 years ago, too

Comment (patricklx) [reply test-projects/gts/package.json:19]: well, it helped reduce some of the `any` types coming from limber imports

Comment (patricklx) [reply test-projects/gts/package.json:19]: mm, maybe we do not need it

**Comment by patricklx** [comment]
i removed the "" prefix. it just means that we cannot parse files that have sequential template definitions. e.g. 
```gjs
<template>A</template>
<template>B</template>
```

**Review by NullVoxPopuli — APPROVED** [review]

**Pull request merged**